File: com/HappyCow/Plugins/ExamplePlugin.java
package com.HappyCow.Plugins;

import com.HappyCow.Plugins.PluginManager.Plugin;
import com.HappyCow.Plugins.PluginManager.PluginMetadata;

// Plugin metadata for discovery.
@PluginMetadata(name = "ExamplePlugin", description = "Simple example plugin.", version = "1.0")

/**
* ExamplePlugin class.
*/
public class ExamplePlugin implements Plugin
{
	/**
	* Function that is executed by PluginManager.
	*/
	@Override
	public void execute()
	{
		System.out.println("Hello, this is a simple plugin!"); // Prints text.
		// Add more variables or function.
	}
}

File: plugins/ExamplePlugin.c
#include <stdio.h>
// Include the plugin header for exporting the plugin details.
#include "../include/plugin.h"

// Function that is called when loading plugins.
void execute()
{
	printf("Big Fat Cat.\n"); // Print text.
	// Add more here...
}

// Export details for plugin discovery.
Plugin plugin =
{
	.name = "ExamplePlugin",
	.description = "Use this as an example!",
	.version = "1.0",
	.execute = execute
};
